Intersections_3/include/CGAL/Intersections_3/internal/Tetrahedron_3_Bounded_3_do_intersect.h

```diff
@@ -161,23 +161,30 @@ do_intersect(const typename K::Tetrahedron_3 &tet,
 }
 
 template <class K>
-inline
-typename K::Boolean
-do_intersect(const typename K::Tetrahedron_3 &tet,
-             const CGAL::Bbox_3 &bb,
-             const K & k)
-{
-  return do_intersect_tetrahedron_bounded(bb, tet, typename K::Point_3(bb.xmin(), bb.ymin(), bb.zmin()), k);
+inline typename K::Boolean do_intersect(const typename K::Tetrahedron_3 &tet,
+                                        const CGAL::Bbox_3 &bb, const K &k) {
+  // Swap arguments. 
+  return do_intersect(bb, tet, k);
 }
 
-  template <class K>
-inline
-typename K::Boolean
-do_intersect(const CGAL::Bbox_3 &bb,
-             const typename K::Tetrahedron_3 &tet,
-             const K & k)
-{
-  return do_intersect_tetrahedron_bounded(bb, tet, typename K::Point_3(bb.xmin(), bb.ymin(), bb.zmin()), k);
+// BBox_3 sphecific code since it's ok for BBox_3 to degenerate.
+template <class K>
+inline typename K::Boolean do_intersect(const CGAL::Bbox_3 &aabb,
+                                        const typename K::Tetrahedron_3 &tet,
+                                        const K &k) {
+  using Tr = CGAL::Triangle_3<K>;
+  typename K::Boolean result = do_intersect(aabb, Tr(tet[0], tet[1], tet[2]), k);
+  if (certainly(result)) return result;
```

Comment:
I'm not sure the `certainly` are useful since if the result if not certain, then the next line will throw.

Reply:
... I agree with Sébastien. `operator||` has no overload for `Uncertain<bool>`, and this the `result` value is cast to `bool`.

The code should do like in `do_intersect_tetrahedron_bounded`:
```c++
typename K::Boolean b = do_intersect(aabb, Tr(tet[0], tet[1], tet[2]), k);
if(certainly(b)) return b;
if(is_indeterminate(b)) result = b;
```